### DISCUSSION TITLE
Tried to pull together a list of C++ JSON parsers to compare them

### DISCUSSION THREAD

**Discussion by Loki-Astari** [nlohmann/json#2309] [Show and tell]
The github project is here:
https://github.com/Loki-Astari/JsonBenchmark

The results of running the tests are here:

https://lokiastari.com/Json/Performance.osx.html
https://lokiastari.com/Json/Performance.linux.html

All test are run on travis (and thus the times are relative to the machines used by travis to give you a relative understanding of performance (not an absolute performance)).

Any suggestions (or help) in making it better (more useful).

**Comment by t-b** [comment]
@Loki-Astari Have a look at https://github.com/nst/JSONTestSuite they have a list of various JSON parsers. I recently got to know https://github.com/simdjson/simdjson which has impressive parsing speeds.

**Reply by Loki-Astari** [reply]
Thanks.
The JSONTestSuite is interesting but has no C++ parsers (which is what I am interested in). But I do like the image of conformance.

I was not aware of simdjson. Since that is C++ I will add it to my test suite.

**Reply by t-b** [reply]
@Loki-Astari The parsers directory has among others rapidjson.

**Reply by Loki-Astari** [reply]
@t-b Thanks.
nlohmann has also pointed it out that I should probably be doing more extensive testing.
There is now an open issue for this: https://github.com/Loki-Astari/JsonBenchmark/issues/3

**Comment by nlohmann** [comment]
Thanks for sharing! This library has a SAX parser - maybe it can be used for the "SaxRoundtrip" test? I'm not sure what this is.
